src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioWorkspaceImpl.cs

```diff
@@ -340,13 +340,20 @@ public override EnvDTE.FileCodeModel GetFileCodeModel(DocumentId documentId)
                 throw new ArgumentNullException(nameof(documentId));
             }
 
-            var documentFilePath = GetFilePath(documentId);
-            if (documentFilePath == null)
+            var document = _threadingContext.JoinableTaskFactory.Run(() => CurrentSolution.GetDocumentAsync(documentId, includeSourceGenerated: true).AsTask());
```

Comment:
:(

is this on UI thread?  does this mean we're now blocking hte UI thread on unbounded source-generator work?

Reply:
That's not new, in essence: any CodeModel operation is going to be asking for semantics, which will block.

Reply:
Chatting with @CyrusNajmabadi offline, we also realized that in many cases they only way you get here is already having ran generators. i.e. you won't have known there's a DocumentId for the source generated file in the first place without generators having ran.